Wangscape/image/TileRearranger.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <array>
+
+#include "Image.h"
+#include "TileRearrangement.h"
+
+class TileRearranger
+{
+public:
+    TileRearranger();
+    ~TileRearranger() = default;
+
+    const TileRearrangement rearrangeTile(const sf::Image& base_tile,
+                                          IVec offset_a,
+                                          IVec offset_b);
+
+    constexpr static unsigned int Corners = 4;
+    std::array<sf::Color, Corners> regionColours;
```

Comment:
Could it be `private`?

Reply:
Eventually `TileRearranger` will be run on user-provided partition images, and enforcing the RGBY colour scheme may harm accessibility. For that reason I think it's important for `regionColours` to be mutable. They could be provided in the constructor or `rearrangeTile()`, but I think it's more appropriate for them to be assigned separately. Since it's a `std::array` and `sf::Color` is just a struct, the only harm that could be done is specifying incorrect colours (non-unique or missing). We check for both of those issues before processing, so they don't really need to be hidden behind `get`/`set` methods.

Reply:
> Since it's a std::array and sf::Color is just a struct, the only harm that could be done is specifying incorrect colours (non-unique or missing).

I sounds like a separate entity - `class` that checks all these preconditions in its constructor.